source/Hardpoint.cpp

```diff
@@ -124,20 +124,20 @@ int Hardpoint::BurstRemaining() const
 
 
 
-// Perform one step (i.e. decrement the reload count).
-void Hardpoint::Step()
+// Perform many steps (i.e. decrement the reload count when the date changes).
+void Hardpoint::Step(int frames)
 {
 	if(!outfit)
 		return;
 	
 	wasFiring = isFiring;
 	if(reload > 0.)
-		--reload;
+		reload -= frames;
```

Comment:
Agreed. I'll hold off on an update for a moment though, pending the conclusion of the discussion.